### ISSUE TITLE
Using -fno-math-errno by default is causing numerical errors

### ISSUE THREAD

**Issue by akesandgren** [easybuilders/easybuild-framework#3598]
GROMACS/2021 with foss/2020a is getting numerical errors due to -fno-math-errno.

-fno-math-errno should NOT be turned on by default since it has numerical impact.
From GCC manual:

https://gromacs.bioexcel.eu/t/test-failure-in-fourcenter-tests-in-gromacs-2021/1633

**Comment by boegel** [comment]
Are these numerical errors caught by the GROMACS tests?

You can avoid the use of `-fno-math-errno` by using `toolchainopts = {'precise': True}` (or `strict`).

**Comment by akesandgren** [comment]
Yes, numerical errors, but only happening with GCC/9.3.0 with -O3 -fno-math-errno

**Comment by boegel** [comment]
Use of `-fno-math-errno` was introduced in #2388 (EasyBuild v3.6.0)

**Comment by boegel** [comment]
@akesandgren Can you summarize the conclusion we reached when discussing this issue during today's EasyBuild conf call?

**Comment by akesandgren** [comment]
We'll leave -fno-math-errno enabled in defaultprec, but leave this issue open as a reminder to keep a eye open for any problems in the future.
The GROMACS problem is a likely a corner case, and only triggers for AVX2 with FMA and may be a problem either in GROMACS or in GCC/9.3.0 or a combination.
